dev.sparcles.SREngine.yml

```diff
@@ -0,0 +1,48 @@
+id: dev.sparcles.SREngine
+runtime: org.freedesktop.Platform
+runtime-version: '23.08'
+sdk: org.freedesktop.Sdk
+command: srengine-run.sh
+
+build-options:
+  build-args:
+    - --share=network
+
+modules:
+   - name: srengine
+     buildsystem: cmake-ninja
+     config-opts:
+       - -DCMAKE_BUILD_TYPE=Debug
+       - -DSR_ENGINE_FLATPAK_BUILD=ON
+       - -DSR_ENGINE_EXTENDED_NAME=OFF
+
+     build-commands:
+       - cmake --build . -j ${FLATPAK_BUILDER_N_JOBS}
+       - install -Dm755 Engine/srengine /app/bin/srengine
+       - mkdir -p /app/share/SREngine
+       - cp -r ../Resources /app/share/SREngine
+     builddir: true
+     sources:
+       - type: git
+         url: https://github.com/SpaRcle-Studio/SREngine
+         branch: master
+
+   - name: script
+     buildsystem: simple
+     build-commands:
+       - install -Dm755 srengine-run.sh /app/bin/srengine-run.sh
+     sources:
+       - type: script
+         dest-filename: srengine-run.sh
+         commands:
+           - /app/bin/srengine -resources /app/share/SREngine/Resources
+
+
+finish-args:
+  - --share=ipc
+  - --socket=x11
+  - --device=dri
+  - --device=input
+  - --socket=pulseaudio
+  - --share=network
+  - --filesystem=host:create
```

Comment:
why host permission? What does `host:create` do?

(it's way too broad)

Reply:
I changed it to home:create, but the engine fails when trying to access and create a cache directory in the /app/share/SREngine/Resources directory which I talked about in the body of the PR.

Reply:
why `:create`? What does mean?

Reply:
Now that I have read it again, I must have misunderstood the flatpak documentation about file system permissions. I changed it to `filesystem=home`, but I experience this issue.
![image](https://github.com/user-attachments/assets/bf0def53-f199-4045-84b4-2059723ec1c7)

Reply:
This path is read only. It's a bad idea, and it's not even specific to flatpak.

Reply:
I have changed the path used. Now I have another problem: org.freedesktop.sdk does not provide a GCC compiler for compiling scripts. Do you have any suggestions on how to fix this?